### PR TITLE
fix: copy failure in NSIS bundler #6743

### PR DESCRIPTION
**Pull request by karandit**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/bundler/src/bundle/windows/nsis.rs [hunk 1]
```diff
@@ -100,6 +100,13 @@ fn get_and_extract_nsis(nsis_toolset_path: &Path, _tauri_tools_path: &Path) -> c
   let data = download(NSIS_APPLICATIONID_URL)?;
   info!("extracting NSIS ApplicationID plugin");
   extract_zip(&data, &nsis_plugins)?;
+  info!("extracted NSIS ApplicationID plugin");
+
+  let nsis_plugins_x86 = nsis_plugins.join("x86-unicode");
+  if !nsis_plugins_x86.exists() {
+    create_dir_all(nsis_plugins_x86)?;
+  }
```
Comment (FabianLars) [inline tooling/bundler/src/bundle/windows/nsis.rs]:
```suggestion
  create_dir_all(nsis_plugins.join("x86-unicode"))?;
```
No need to check if it exists first imo. create_dir_all won't delete anything if a dir already exists.

**Review by FabianLars — CHANGES_REQUESTED** [review]
Thank you! I could swear i tested this when i first added Linux support but glad you cought it before it went into v1.3.

I added just one small code change request below. Also i'm not reallyyy a fan of the new log just because we usually don't log the completion 😅 idk, just something to think about if we might wanna remove that again for now.

File: tooling/bundler/src/bundle/windows/nsis.rs (hunk 1 above)
Comment (karandit) [reply tooling/bundler/src/bundle/windows/nsis.rs]: thank you, I'm not a Rust developer .... yet :)

**Comment by karandit** [comment]
It's my  honor to contribute.
Yeah, it's tricky. I've also observed, during my experiments, that it worked if it was cached once

**Comment by karandit** [comment]
let me remove that extra log line and squash it again

**Comment by FabianLars** [comment]
Don't worry about squashing, we'll do that when that PR gets merged anyway :) (feel free to do it though)

**Comment by karandit** [comment]
btw, I wanted to remove the duplication of this: `nsis_plugins.join("x86-unicode")` which appears in 3 places now, but I bumped into the borrow checker and I thought that maybe I leave it to the pros

**Review by FabianLars — APPROVED** [review]

**Comment by FabianLars** [comment]
i tend to say that i prefer the duplicates because i can identify the paths quicker this way and the bundler doesn't need to be overly optimized either. So i'll leave that to the others as well :P

**Pull request merged**